cloud-core/src/main/java/cloud/commandframework/arguments/standard/DurationArgument.java

```diff
@@ -0,0 +1,269 @@
+            final @NonNull ArgumentDescription defaultDescription
+    ) {
+        super(
+                required,
+                name,
+                new DurationArgument.DurationParser<>(),
+                defaultValue,
+                Duration.class,
+                suggestionsProvider,
+                defaultDescription
+        );
+    }
+
+    /**
+     * Create a new builder
+     *
+     * @param name Name of the component
+     * @param <C>  Command sender type
+     * @return Created builder
+     */
+    public static <C> @NonNull Builder<C> newBuilder(final @NonNull String name) {
+        return new Builder<>(name);
+    }
+
+    /**
+     * Create a new required command component
+     *
+     * @param name Component name
+     * @param <C>  Command sender type
+     * @return Created component
+     */
+    public static <C> @NonNull CommandArgument<C, Duration> of(final @NonNull String name) {
+        return DurationArgument.<C>newBuilder(name).asRequired().build();
+    }
+
+    /**
+     * Create a new optional command component
+     *
+     * @param name Component name
+     * @param <C>  Command sender type
+     * @return Created component
+     */
+    public static <C> @NonNull CommandArgument<C, Duration> optional(final @NonNull String name) {
+        return DurationArgument.<C>newBuilder(name).asOptional().build();
+    }
+
+    /**
+     * Create a new required command component with a default value
+     *
+     * @param name            Component name
+     * @param defaultDuration Default duration
+     * @param <C>             Command sender type
+     * @return Created component
+     */
+    public static <C> @NonNull CommandArgument<C, Duration> optional(
+            final @NonNull String name,
+            final @NonNull String defaultDuration
+    ) {
+        return DurationArgument.<C>newBuilder(name).asOptionalWithDefault(defaultDuration).build();
+    }
+
+
+    public static final class Builder<C> extends CommandArgument.Builder<C, Duration> {
+
+        private Builder(final @NonNull String name) {
+            super(Duration.class, name);
+        }
+
+        /**
+         * Builder a new boolean component
+         *
+         * @return Constructed component
+         */
+        @Override
+        public @NonNull DurationArgument<C> build() {
+            return new DurationArgument<>(this.isRequired(), this.getName(), this.getDefaultValue(),
+                    this.getSuggestionsProvider(), this.getDefaultDescription()
+            );
+        }
+
+    }
+
+
+    public static final class DurationParser<C> implements ArgumentParser<C, Duration> {
+
+        @Override
+        public @NonNull ArgumentParseResult<Duration> parse(
+                final @NonNull CommandContext<C> commandContext,
+                final @NonNull Queue<String> inputQueue
+        ) {
+            final String input = inputQueue.peek();
```

Comment:
You could use poll here, than you don't need to remove agterwards..

Reply:
You should only remove from the input queue upon successful parsing, removing from the queue on failure will cause various problems.

Reply:
Yes, but peek -> null check -> remove is the same thing than poll -> null check.. There is just no way this fails in between

Reply:
Yes, because the code should not be calling remove there, it should be removing from the queue upon successful parsing, as per my comment.

Reply:
I'm dum nevermind then

Reply:
Fixed this to call `inputQueue.remove()` on line 193. Thank you